subprojects/language-java/src/integTest/groovy/org/gradle/java/compile/incremental/GroovyJavaJointIncrementalCompilationIntegrationTest.groovy

```diff
@@ -63,11 +63,11 @@ class GroovyJavaJointIncrementalCompilationIntegrationTest extends AbstractJavaG
         where:
         scenario                                    | initialSet            | firstChange                   | firstBuildMessage                                        | secondChange                  | secondBuildMessage
         'Add Java files to Groovy file set'         | ['G']                 | ['G', 'J']                    | 'non-Groovy file'                                        | ['G', 'J']                    | 'UP-TO-DATE'
-        'Add Groovy files to Java file set'         | ['J']                 | ['G', 'J']                    | 'unable to get source-classes mapping'                   | ['G', 'J.changed']            | 'non-Groovy file'
+        'Add Groovy files to Java file set'         | ['J']                 | ['G', 'J']                    | 'G.groovy has been added'                                | ['G', 'J.changed']            | 'non-Groovy file'
```

Comment:
This is now fixed because previously we had no distinction between "no mapping" and "empty mapping". Now we do with the `CompilerApiData.isAvailable`